src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,37 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
+
+    public final String value;
+    private boolean isPrivate;
+  
+    /**
+     * Validates given contact.
+     *
+     * @throws IllegalValueException if given contact string is invalid.
+     */    
+    public Contact(String contact, boolean isPrivate) throws IllegalValueException {
+        this.isPrivate = isPrivate;
+        this.value = contact.trim();
+    }
+    
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
```

Comment:
recollect the discussion in tutorial why this may not be the most accurate?
by virtue of inheritance, address and email are all contacts; hence what happens?